### PR TITLE
Proof of concept

### PR DESCRIPTION
**Pull request by JE-FH**
@MJHC review please

### CODE REVIEW

File: src/lib/Commands/ExecuteCommand.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        let input_clone = String::from(input);
+        let mut stdin = process.stdin.take().expect("failed to get stdin");
+        std::thread::spawn(move || {
+            stdin.write_all(input_clone.as_bytes()).expect("failed to write to stdin");
+        });
+
+        let output = process
+            .wait_with_output()
+            .expect("penis");
```
Comment (MJHC) [inline src/lib/Commands/ExecuteCommand.rs]:
```suggestion
            .expect("benis");
```

**Review by MJHC — CHANGES_REQUESTED** [review]
Crashes when inputting something that is not a program ie "hello" world panics. "ls" works fine. 

And shouldn't the interfaces be named traits because it is in rust? 

There is also a lot of uneccessary parantheses in the if statements. If you do not care about warnings, then its fine. 

should probably fix the unexpected panic.

File: src/lib/Commands/ExecuteCommand.rs (hunk 1 above)
Comment (JE-FH) [reply src/lib/Commands/ExecuteCommand.rs]: penis to benis suggestion is a good suggestion however interfaces are not interfaces per say, they just interface with terminal, so interface in this context does not mean OOP like interface but more of interacting with another thing like terminal. 

Warnings about code style is in my opinion stupid and only warnings impacting semantics will be considered.

I will also fix panics

**Comment by JE-FH** [comment]
after many hours of work this should be kinda working, but only on windows, a linux or unix command resolver would be needed to get it working on mac or linux

**Review by MJHC — APPROVED** [review]
bitmoji

**Pull request merged**